src/app/archive/archive_lib/processor.ml

```diff
@@ -111,6 +111,1078 @@ module Public_key = struct
           public_key
 end
 
+module Snapp_state_data = struct
+  let find (module Conn : CONNECTION) (fp : Pickles.Backend.Tick.Field.t) =
+    Conn.find
+      (Caqti_request.find Caqti_type.string Caqti_type.int
+         "SELECT id FROM snapp_state_data WHERE field = ?")
+      (Pickles.Backend.Tick.Field.to_string fp)
+
+  let find_opt (module Conn : CONNECTION) (fp : Pickles.Backend.Tick.Field.t) =
+    Conn.find_opt
+      (Caqti_request.find_opt Caqti_type.string Caqti_type.int
+         "SELECT id FROM snapp_state_data WHERE field = ?")
+      (Pickles.Backend.Tick.Field.to_string fp)
+
+  let add_if_doesn't_exist (module Conn : CONNECTION)
+      (fp : Pickles.Backend.Tick.Field.t) =
+    let open Deferred.Result.Let_syntax in
+    match%bind find_opt (module Conn) fp with
+    | Some id ->
+        return id
+    | None ->
+        Conn.find
+          (Caqti_request.find Caqti_type.string Caqti_type.int
+             "INSERT INTO snapp_state_data (field) VALUES (?) RETURNING id")
+          (Pickles.Backend.Tick.Field.to_string fp)
+end
+
+module Snapp_state_data_array = struct
+  let add_if_doesn't_exist (module Conn : CONNECTION)
+      (_fps : Pickles.Backend.Tick.Field.t array) =
+    (*
+    let open Deferred.Result.Let_syntax in
+    let element_ids_typ = vector n Caqti_type.int in
+    let%bind (element_ids : (int, 'n) Vector.vec) =
```

Comment:
based on https://www.postgresql.org/docs/9.1/arrays.html it seems like we can use `Caqti_type.string` to represent array values when inserting. In this case it'd be `element_ids= '{2,3,4,..}'` and when retrieving, we could parse the same string to an ocaml array